go/database/mpt/forest.go

```diff
@@ -383,6 +383,24 @@ func (s *Forest) SetValue(rootRef *NodeReference, addr common.Address, key commo
 	return newRoot, err
 }
 
+func (s *Forest) HasEmptyStorage(rootRef *NodeReference, addr common.Address) (isEmpty bool, err error) {
+	v := &lambdaVisitor{
+		visit: func(node Node, info NodeInfo) VisitResponse {
+			isEmpty = node.(*AccountNode).storage.Id().IsEmpty()
```

Comment:
this visitor provides all nodes on the path. I.e. it should check that the cast is possible that the node is of the `AccountNode` type.  The tests must be improved if they pass at the moment.